### DISCUSSION TITLE
Access Media Azure files in MVC application without CMS

### DISCUSSION THREAD

**Discussion by vsezima** [OrchardCMS/OrchardCore#14582] [Q&A]
Hello,
I'm having trouble accessing Azure media files in MVC application without CMS (created with this tutorial https://docs.orchardcore.net/en/latest/docs/guides/create-modular-application-mvc/) and was wondering, if it is even possible.

When I create CMS website (with this tutorial https://docs.orchardcore.net/en/latest/docs/guides/create-cms-application/) and add Azure media feature and proper appsettings configuration, then I see Azure files in CMS media library on URL /admin/media and can access them directly on URL /media/filename.

For MVC application, I added needed references and appsettings, but cannot access same file on URL /media/filename (it returns 404).
.csproj references:
![image](https://github.com/OrchardCMS/OrchardCore/assets/15254338/c8161332-6466-48fb-bcad-55c0d54a8821)
Appsettings configuration:
![image](https://github.com/OrchardCMS/OrchardCore/assets/15254338/0b920aa4-532c-4929-ac2e-e80897fb639d)
Program.cs:
![image](https://github.com/OrchardCMS/OrchardCore/assets/15254338/e339e05a-ce6b-4885-9e18-547953f0a9a0)

In Program.cs I had to add 3 extra extensions methods so URL /media/filename wouldn't throw exceptions about missing services, so I think that Media feature works, but it still can't find requested file.

**Comment by MikeAlhayek** [comment]
Can you show your code? How are you accessing the media file? What code is failing? what do you expect to happen.

**Reply by vsezima** [reply]
For reproducible MVP there isn't really any more code than on those screenshots.
I just did following:

When I create default CMS app (with this tutorial https://docs.orchardcore.net/en/latest/docs/guides/create-cms-application/, site recipe for example Blog or Blank), media library is empty at first (on URL https://localhost:5001/Admin/Media).
Then I enable feature Azure Media storage with correct settings in appsettings.json and see my files from Azure in media library and can access them on URL https://localhost:5001/media/test_sample.png
With CMS this works, but I don't want CMS, only standalone MVC.

So then I try do the same for default MVC app without CMS (with this tutorial https://docs.orchardcore.net/en/latest/docs/guides/create-modular-application-mvc/). 
Since there is no Admin, I try to enable Azure Media storage through .csproj references and appsettings.json.
That works, but threw few exceptions (when trying to access file on URL https://localhost:5001/media/test_sample.png) about missing services implementations, so I added them in Program.cs with extensions methods AddDataAccess(), AddDataStorage() and AddDocumentManagement().
Now the URL https://localhost:5001/media/test_sample.png no longer throw exceptions but returns only 404 Not Found, which I'm not sure if it means that it cannot find that file "test_sample.png" or that endpoint /media.

I expect that the endpoint /media is managed through StaticFiles set with appsettings options here https://github.com/OrchardCMS/OrchardCore/blob/8d2aa01a757a3cc8a99851cff785ac87862f1c4a/src/OrchardCore.Modules/OrchardCore.Media/Startup.cs#L221 and replaced with Azure Media storage here when enabled https://github.com/OrchardCMS/OrchardCore/blob/8d2aa01a757a3cc8a99851cff785ac87862f1c4a/src/OrchardCore.Modules/OrchardCore.Media.Azure/Startup.cs#L92
so I expect it should work without CMS, with only these modules enabled.

There is also this Info log about missing action description, so I'm not sure what else to enable so that endpoint /media works:
![image](https://github.com/OrchardCMS/OrchardCore/assets/15254338/190ae253-611f-4527-a0cb-3f7c6ab7fe67)

**Reply by MikeAlhayek** [reply]
I am never tried using it outside the CMS.  But, you may want to investigate the these were needed. Maybe check the exception and see which service was not registered. I am guesssing ISiteService somewhere around the line which may need to be fixed.

![image](https://github.com/OrchardCMS/OrchardCore/assets/24724371/1942394d-6c38-4dc6-a9a4-182a9da656a8)

In regard to your issues, maybe the staticFileProvider is not registered correctly in the pipeline.
